build/common.targets

```diff
@@ -309,4 +309,15 @@
     </ItemGroup>
   </Target>
 
+  <!--
+    ============================================================
+    EnsureNewtonsoftJsonVersion ensures that the resolved version of Newtonsoft.Json is the version that ships with VS/.NET Core SDK
+    ============================================================
+  -->
+  <Target Name="EnsureNewtonsoftJsonVersion" AfterTargets="ResolveAssemblyReferences">
+    <Error
+      Text="Newtonsoft.Json must be version $(NewtonsoftJsonVersionCore) but resolved %(Reference.NuGetPackageVersion)"
+      Condition=" %(Reference.NuGetPackageId) == 'Newtonsoft.Json' AND %(Reference.NuGetPackageVersion) != '$(NewtonsoftJsonVersionCore)' " />
```

Comment:
- Does `Reference.NuGetPackageId` will not throw a null pointer exception?
- Does `%` always resolve to a single element ?

Reply:
MSBuild doesn't seem to have the concept of null. Any property or item metadata not defined just has the value of an empty string.

I don't understand MSBuild `%` or `@` very well. I know when it comes to messages or `<Exec>`, `%` causes each match to be run separately, whereas `@` causes all matches to be concatenated into a single string. I wanted each Reference item to be considered separately in my condition, so I tried using `%` and it seemed to work.